### PR TITLE
Add GraalVM native image demo for agent support

### PR DESCRIPTION
**Pull request by ziyilin**

### CODE REVIEW

File: graalvm-native-image-demo/spring-boot-ot/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+### Prequisite
+
+ - Build the `agent support PR` to get the GraalVM JDK. 
+ - Or download the [dev version binary](https://github.com/ziyilin/ziyi-forked-graal/releases/tag/static-instrument) for experiment only
+ - Make sure your `GRAALVM_HOME` system variable refers to the proper GraalVM
+
+### Steps
+1. cd to the root of this project
+2. `mvn package`: Build the jar package of this project
+3. `./run.sh --jvm --collect --log`: Execute the project in JVM mode, enabling OpenTelemetry agent to log monitoring data. Meanwhile, native-image-agent is enbeled to collect transformation classes and other dynamic data. When this step ends, data is dumped to `native-configs` directory.
```
Comment (steverao) [inline graalvm-native-image-demo/spring-boot-ot/README.md:23]:
```suggestion
3. `./run.sh --jvm --collect --log`: Execute the project in JVM mode, enabling OpenTelemetry agent to log monitoring data. Meanwhile, native-image-agent is enabled to collect transformation classes and other dynamic data. When this step ends, data is dumped to `native-configs` directory.
```

File: graalvm-native-image-demo/spring-boot-ot/run.sh [hunk 2]
```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+JAVA="$HOME/tools/GRAALVM21"
```
Comment (steverao) [inline graalvm-native-image-demo/spring-boot-ot/run.sh:2]: The line is correct? If users set `export GRAALVM_HOME=/root/graalvm-test/graalvm-17-23.0.2-ali-1.1-instru` according to README.md, it seems they can't locate to relevant directory.

**Review by steverao — COMMENTED** [review]
BTW, I suggest that use `graalvm-demo` to replace the `graalvm-native-image-demo` and create a directory named `ot-demo` under `graalvm-demo`. then, replace `spring-boot-ot` by `spring-boot`. In this way, it looks like more clearly and we can add other frameworks demo easily.

**Comment by ziyilin** [comment]
Graalvm-demo is not accurate. Because Graalvm could be Graalvm JIT or Graalvm native image. Graalvm-native-image is more accurate.

**Pull request merged**